qiskit/visualization/pulse/matplotlib.py

```diff
@@ -584,7 +584,12 @@ def draw(self, schedule, dt, interp_method, plot_range,
             # when input schedule is empty or comprises only frame changes,
             # we need to overwrite pulse duration by an integer greater than zero,
             # otherwise waveform returns empty array and matplotlib will be crashed.
-            tf = schedule.stop_time or 1
+            # tf = schedule.stop_time or 1
+            tf = 1
+            for chan in channels_to_plot:
```

Comment:
Could this be something like

```python
if channels_to_plot:
    tf = schedule.timeslots.ch_duration(*channels_to_plot)
else:
    tf = schedule.stop_time
tf = tf or 1
```

Reply:
Yeah, this looks much cleaner. I will make the change.